lib/membrane/mpeg/ts/demuxer.ex

```diff
@@ -27,7 +27,21 @@ defmodule Membrane.MPEG.TS.Demuxer do
 
   def_output_pad(:output,
     availability: :on_request,
-    accepted_format: %module{} when module in [Membrane.RemoteStream, Membrane.H264]
```

Comment:
@varsill what if we determine which streams are aligned and provide the correct output format as you did w/o requiring the user to specify the expected alignment? In my case for example, I do not always know it beforehand, while this code is in the correct position to find it out. Eventually the pad format checks are going to scream at you anyway if for example alignment is required on output and the input does not provide it. WDYT?

Reply:
I've got your point @dmorn, I thought about it initially and I've even tried to write it that way, however I found the following scenario problematic: 
* It might happen, that at first the PES are "AU-aligned" and then suddenly an AU split among two PESs arrives. In such a case, the alignment_indicator of the first PES that contains that split AU will still have the `alignment_indicator=1` and we will need to wait for the next PES to arrive (which has `alignment_indicator=0`) before sending the first PES because otherwise we won't be able to properly update the stream format.

I had an impression that adding such a "buffering" heavily affects the element's architecture and that's why I have proposed a solution with option passing.

Reply:
Thank you for the contribution @varsill 😉 I see, but this makes me even more convinced that the user should not be responsible for setting the option! In fact, if the stream format suddenly changes, the option set by the user will be correct just for a portion of the stream. What if we deliver another stream_format message in that case? I believe this is the nature of TS, isn't it? Isn't it related to the discontinuity indicator found in HLS streams? Maybe I'm not getting this right, in fact this is the reason we did not support this case both here and in our HLS plugin :D

I am aware that this does not make things easy for plugins attached to the output pads, but I still do not see setting the option beforehand a solution! What happens in that case in your example stream with this solution?

Reply:
Oh, I see - actually in case of my example stream, with `is_aligned: true` option the `Membrane.H264{alignment: :au}` stream format will be delivered before the first buffer and later, once the `alignment_indicator` value changes, the warning will be printed, persuading the user to change the option to `is_aligned: false` because the stream format might not describe the real incoming stream. 
With `is_aligned: false`, the `Membrane.RemoteStream` will be sent as a stream format and the user will be aware, that if some further element in the pipeline requires `Membrane.H264`, the `Membrane.H264.Parser` is needed to be put between the demuxer and that element.
Briefly speaking, what I had in mind was to make sure that further elements in the pipeline won't get misled by the stream format. 

However, I can try to rewrite the demuxer so that to automatically detect the change in alignment and in such a case - send updated stream format.

Reply:
> Briefly speaking, what I had in mind was to let user manually specify what stream format should be send, so that further elements in the pipeline won't get misled by the stream format in case the default stream format is incorrect.

I get your point, but for us at least this would be a no-go as we're also ingesting HLS streams we've no control over: this means that we have no way of knowing this in advance and once the pipeline is running, nobody is going to look at the warning (till its too late). The next question before proceeding would be: what elements are required to handle both cases? What do you attach to the output pads of these streams? We need a solution that works anyway I guess, every user of this plugin will have the same problem!

Reply:
Hi @dmorn , I have implemented the discussed approach of updating the stream format based on the values read from the field. 
Because demuxer is usually put right before parser, I have also added support for changing of mode of H264 parser, since it has been missing it since now: https://github.com/membraneframework/membrane_h264_plugin/pull/50

Reply:
Great work @varsill!